src/scenes/Home/Home.tsx

```diff
@@ -1,51 +1,60 @@
-import { RouteComponentProps } from '@reach/router';
-import { memo, useCallback, useState } from 'react';
+import { navigate, Redirect, RouteComponentProps } from '@reach/router';
+import { memo, useCallback } from 'react';
 import styled from 'styled-components';
-import Password from 'components/Password/Password';
+import useIsPasswordValidated from 'hooks/useIsPasswordValidated';
+import useIsAuthenticated from 'contexts/auth/useIsAuthenticated';
 import { Text } from 'components/core/Text/Text';
-import WalletSelector from 'components/WalletSelector/WalletSelector';
+import Spacer from 'components/core/Spacer/Spacer';
+import Button from 'components/core/Button/Button';
 
 function Home(_: RouteComponentProps) {
-  // on dev, this will route to localhost:4000/api/test
-  // on prod, this will route to api.gallery.so/api/test
-  // const { data, error } = useSwr('/test');
-  // console.log('the result', data, error);
-
-  const [showWalletSelector, setShowWalletSelector] = useState(false);
-  const handleNextClick = useCallback(() => {
-    setShowWalletSelector(true);
+  // whether the user has entered the correct password
+  const isPasswordValidated = useIsPasswordValidated();
+  // whether the user is web3-authenticated
+  const isAuthenticated = useIsAuthenticated();
+  // TODO: useIsAuthenticated should return the username (or wallet address)
+  // if indeed authenticated - we should redirect there
+  const username = '0x70d04384b5c3a466ec4d8cfb8213efc31c6a9d15';
+
+  const handleRedirectToProfile = useCallback(() => {
+    navigate(`/${username}`);
   }, []);
 
-  if (showWalletSelector) {
-    return (
-      <StyledHome>
-        <WalletSelector />
-      </StyledHome>
-    );
+  if (!isPasswordValidated) {
+    return <Redirect to="/password" />;
+  }
+
+  if (!isAuthenticated) {
```

Comment:
I don't think we should auto redirect to the auth step, the user should see the "enter" button

<img width="798" alt="Screen Shot 2021-05-08 at 11 27 01 AM" src="https://user-images.githubusercontent.com/80802871/117522819-5354a480-aff0-11eb-986e-87ad641d0cd9.png">